### PR TITLE
Tickets/preops 3792

### PR DESCRIPTION
**Pull request by galaxyumi**
A new NB tutorial to describe how to use user-supplied tables in DP0.3 queries.

### CODE REVIEW

File: DP03_05_Using_User-Supplied_Catalogs.ipynb [hunk 1]
```diff
@@ -0,0 +1,683 @@
+{
```
Comment (christinawilliams) [inline DP03_05_Using_User-Supplied_Catalogs.ipynb:1]: <p>I would maybe say "Get the path to the home directory" instead? Also above, in the header for section 1.2</p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/rubin-dp0/tutorial-notebooks/pull/187/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='13'/>

Comment (christinawilliams) [inline DP03_05_Using_User-Supplied_Catalogs.ipynb:1]: <p>I think the label for header 2 could be more descriptive. Perhaps something like "Loading a user-supplied table of..."</p><p> </p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/rubin-dp0/tutorial-notebooks/pull/187/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='15'/>

Comment (christinawilliams) [inline DP03_05_Using_User-Supplied_Catalogs.ipynb:1]: <p>This cell had some errors from flake-8, attaching here (after INFO: firstnumber:secondnumber are as: firstnumber = line number, secondnumber = starting character number in line</p>

<pre class="ql-syntax" spellcheck="false">pycodestyle INFO: 2:78: W291 trailing whitespace
pycodestyle INFO: 3:66: W291 trailing whitespace
pycodestyle INFO: 4:80: E501 line too long (105 &gt; 79 characters)
pycodestyle INFO: 47: E203 whitespace before ':'
</pre><p> </p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/rubin-dp0/tutorial-notebooks/pull/187/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='19'/>

Comment (christinawilliams) [inline DP03_05_Using_User-Supplied_Catalogs.ipynb:1]: <p>maybe good to add what the expectation is (that as expected, all the deltas are &lt; 10"?)</p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/rubin-dp0/tutorial-notebooks/pull/187/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='20'/>

Comment (christinawilliams) [inline DP03_05_Using_User-Supplied_Catalogs.ipynb:1]: <p>I got a weird output from this cell, is it indicating some kind of text error?</p>

<pre class="ql-syntax" spellcheck="false">Text(0, 0.5, '$\\Delta$dec [deg]')
</pre><p> </p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/rubin-dp0/tutorial-notebooks/pull/187/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='21'/>

Comment (christinawilliams) [inline DP03_05_Using_User-Supplied_Catalogs.ipynb:1]: <p>Maybe its useful here to have a brief sentence explaining what format for the table is expected by the TAP service, or what formats are generally allowed? It looks like you are reading it in using pandas and then converting it to an astropy table, but does it have to be an astropy table? Or can it just be arrays of values? Wondering if the pandas step us necessary or if one could just use astropy ascii.read in 1 line.</p><p> </p><p>Generally it might be nice to start Section 3 with an overview sentence of the purpose or advantage of this feature. At the moment I focus on ADQL part of the title and I didn't exactly understand what part of this is ADQL (just thinking back to your slack question where all ADQL inner joins are just joins?)</p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/rubin-dp0/tutorial-notebooks/pull/187/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='23'/>

Comment (christinawilliams) [inline DP03_05_Using_User-Supplied_Catalogs.ipynb:1]: <p>flake8 error from this cell:</p>

<pre class="ql-syntax" spellcheck="false">pycodestyle INFO: 47: E203 whitespace before ':'
</pre><p> </p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/rubin-dp0/tutorial-notebooks/pull/187/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='25'/>

Comment (christinawilliams) [inline DP03_05_Using_User-Supplied_Catalogs.ipynb:1]: <p>flake8 error for this cell:</p>

<pre class="ql-syntax" spellcheck="false">pycodestyle INFO: 3:80: E501 line too long (84 &gt; 79 characters)
</pre><p> </p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/rubin-dp0/tutorial-notebooks/pull/187/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='28'/>

Comment (christinawilliams) [inline DP03_05_Using_User-Supplied_Catalogs.ipynb:1]: <p>Maybe somewhere in here it would be useful to have more explanation of what will happen next (i.e. this is still some kind of external catalog that is uploaded to RSP for matching and other analysis, but now you're demonstrating how to grab it from some other server? Anyway some more general explanation of why this is useful could be helpful. Also, perhaps list 1-2 other examples of what catalogs could be accessed by PyVO?</p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/rubin-dp0/tutorial-notebooks/pull/187/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='30'/>

Comment (christinawilliams) [inline DP03_05_Using_User-Supplied_Catalogs.ipynb:1]: <p>I think adequacy maybe isn't the right word to use here but am not sure which word should be used instead. Perhaps just remove ensure adequacy?</p><p> </p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/rubin-dp0/tutorial-notebooks/pull/187/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='32'/>

Comment (christinawilliams) [inline DP03_05_Using_User-Supplied_Catalogs.ipynb:1]: <p>this cell had the following flake8 errors:</p>

<pre class="ql-syntax" spellcheck="false">pycodestyle INFO: 4:63: W291 trailing whitespace
pycodestyle INFO: 6:28: W291 trailing whitespace
pycodestyle INFO: 7:80: E501 line too long (129 &gt; 79 characters)
</pre><p> </p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/rubin-dp0/tutorial-notebooks/pull/187/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='33'/>

Comment (christinawilliams) [inline DP03_05_Using_User-Supplied_Catalogs.ipynb:1]: <p>As a non-solar-system person I didn't understand what these plots are supposed to demonstrate. Maybe a sentence that explains what these plots mean or indicate is useful?</p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/rubin-dp0/tutorial-notebooks/pull/187/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='34'/>

Comment (christinawilliams) [inline DP03_05_Using_User-Supplied_Catalogs.ipynb:1]: <p>this cell had the following flake8 errors:</p>

<pre class="ql-syntax" spellcheck="false">pycodestyle INFO: 4:80: E501 line too long (80 &gt; 79 characters)
pycodestyle INFO: 9:80: E501 line too long (87 &gt; 79 characters)
</pre><p> </p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/rubin-dp0/tutorial-notebooks/pull/187/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='35'/>

Comment (christinawilliams) [inline DP03_05_Using_User-Supplied_Catalogs.ipynb:1]: <p>captial -&gt; capital. Also I think "in some Gaia table" should be "in some Gaia tables" or "in the Gaia table"? But I'm not sure I understood for sure if this correction is needed, I've never used the Gaia table(s) before</p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/rubin-dp0/tutorial-notebooks/pull/187/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='36'/>

Comment (christinawilliams) [inline DP03_05_Using_User-Supplied_Catalogs.ipynb:1]: <p>This cell had the following flake8 error</p>

<pre class="ql-syntax" spellcheck="false">pycodestyle INFO: 1:80: E501 line too long (81 &gt; 79 characters)
</pre><p> </p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/rubin-dp0/tutorial-notebooks/pull/187/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='37'/>

Comment (christinawilliams) [inline DP03_05_Using_User-Supplied_Catalogs.ipynb:1]: <p>Does this need its own cell? Could just print from the earlier cell once you do the denomination fix</p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/rubin-dp0/tutorial-notebooks/pull/187/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='38'/>

Comment (christinawilliams) [inline DP03_05_Using_User-Supplied_Catalogs.ipynb:1]: <p>I think maybe to get the correct formatting on my_home_dir you need ` and not ' ?</p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/rubin-dp0/tutorial-notebooks/pull/187/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='39'/>

Comment (christinawilliams) [inline DP03_05_Using_User-Supplied_Catalogs.ipynb:1]: <p>I'm not sure I understood this point. Is this referring to reading in the catalog in the above cell, or below cell (or is this not actually demonstrated?) Maybe this section 4.2 needs an introductory sentence explaining more about what is about to be demonstrated where it could be explained before doing anything. Also the introductory sentence could say i.e. how is this different from the user-uploaded table from section 2? is it just a different format? </p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/rubin-dp0/tutorial-notebooks/pull/187/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='43'/>

Comment (christinawilliams) [inline DP03_05_Using_User-Supplied_Catalogs.ipynb:1]: <p>flake8 complained about the following:</p>

<pre class="ql-syntax" spellcheck="false">pycodestyle INFO: 2:47: W291 trailing whitespace
pycodestyle INFO: 47: E203 whitespace before ':'
pycodestyle INFO: 13:80: E501 line too long (81 &gt; 79 characters)
</pre><p> </p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/rubin-dp0/tutorial-notebooks/pull/187/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='45'/>

Comment (christinawilliams) [inline DP03_05_Using_User-Supplied_Catalogs.ipynb:1]: <p>I got a negative ID in my run. Did we ever figure out why there were some negative IDs? Maybe we need to explain this in case some people retrieve negative IDs  </p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/rubin-dp0/tutorial-notebooks/pull/187/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='47'/>

Comment (christinawilliams) [inline DP03_05_Using_User-Supplied_Catalogs.ipynb:1]: <p>1) Ocasionally -&gt; Occasionally</p><p>2) I think we are supposed to avoid writing in the first person (or at least that is how the current tutorial notebook guidelines are written) so maybe rephrase "you could run" -&gt; the following error message may occur. Then later "lets get an instance" -&gt; "get an instance"</p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/rubin-dp0/tutorial-notebooks/pull/187/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='48'/>

Comment (christinawilliams) [inline DP03_05_Using_User-Supplied_Catalogs.ipynb:1]: <p>flake8 errors here:</p>

<pre class="ql-syntax" spellcheck="false">pycodestyle INFO: 2:28: W291 trailing whitespace
pycodestyle INFO: 3:33: W291 trailing whitespace
</pre><p> </p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/rubin-dp0/tutorial-notebooks/pull/187/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='50'/>

Comment (christinawilliams) [inline DP03_05_Using_User-Supplied_Catalogs.ipynb:1]: <p>UTC and MJD should be defined on first use</p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/rubin-dp0/tutorial-notebooks/pull/187/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='51'/>

Comment (christinawilliams) [inline DP03_05_Using_User-Supplied_Catalogs.ipynb:1]: <p>I think it should be "plot the"?</p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/rubin-dp0/tutorial-notebooks/pull/187/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='53'/>

Comment (christinawilliams) [inline DP03_05_Using_User-Supplied_Catalogs.ipynb:1]: <p>flake8 errors for this cell:</p>

<pre class="ql-syntax" spellcheck="false">pycodestyle INFO: 2:57: W291 trailing whitespace
pycodestyle INFO: 4:58: W291 trailing whitespace
pycodestyle INFO: 6:38: E231 missing whitespace after ','
pycodestyle INFO: 7:38: E231 missing whitespace after ','
</pre><p> </p><p> </p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/rubin-dp0/tutorial-notebooks/pull/187/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='54'/>

Comment (galaxyumi) [reply DP03_05_Using_User-Supplied_Catalogs.ipynb]: <p>I think this is int64 thing and we don't need to explain in this particular NB?</p>

Comment (christinawilliams) [reply DP03_05_Using_User-Supplied_Catalogs.ipynb]: Ok! Whatever you think is best, I just flagged because I remember people asking about this but its probably fine.

Comment (galaxyumi) [reply DP03_05_Using_User-Supplied_Catalogs.ipynb]: <p>Definitely a pandas table is not working. I made this part simpler, by not converting the result table to pandas dataframe in the first place. Astropy tables seem to the expected table format, but if a table includes unicode column, it is not acceptable. I am trying to understand some details, but it's something deeper in the database. Need to ask some help from the dm_rsp team. </p><p> </p><p>I'm not sure if I understand your comment about ADQL - I copied this title from your portal tutorial. This part is to join a user-uploaded table to a DP0.3 table using ssObjectId. Do you think the title needs to be something else?</p><p> </p><p> </p><p> </p>

Comment (galaxyumi) [reply DP03_05_Using_User-Supplied_Catalogs.ipynb]: <p>The only difference between the tables in Sections 2,3 and the table in this section is that there are string columns. I am also not sure what's the requirement for the uploading table format, as I responded to one of your earlier comments. I figured out a solution based on the error msg when trying to upload a table including columns in the unicode format, but without a full understanding what's going on here.</p><p> </p>

Comment (galaxyumi) [reply DP03_05_Using_User-Supplied_Catalogs.ipynb]: <p>Added some explanation</p>

Comment (galaxyumi) [reply DP03_05_Using_User-Supplied_Catalogs.ipynb]: <p>Added a sentence to improve the clarity</p><p> </p>

File: DP03_06_Using_User-Supplied_Catalogs.ipynb [hunk 2]
```diff
@@ -0,0 +1,686 @@
+{
```
Comment (christinawilliams) [inline DP03_06_Using_User-Supplied_Catalogs.ipynb:1]: <p>Would rephrase that "The TAP service does not accept uploads in a Pandas DataFrame format. Thus..."</p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/rubin-dp0/tutorial-notebooks/pull/187/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='40'/>

Comment (christinawilliams) [inline DP03_06_Using_User-Supplied_Catalogs.ipynb:1]: <p>Typo: individual</p><p> </p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/rubin-dp0/tutorial-notebooks/pull/187/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='47'/>

**Comment by galaxyumi** [comment]
<p>All the suggestions and typos were corrected. Thank you, Christina!</p>
